lib/new_relic/agent/instrumentation/redis/instrumentation.rb

```diff
@@ -10,7 +10,9 @@ module Redis
     LOCALHOST = 'localhost'
     MULTI_OPERATION = 'multi'
     PIPELINE_OPERATION = 'pipeline'
-    HAS_REDIS_CLIENT = Gem::Version.new(::Redis::VERSION) >= Gem::Version.new('5.0.0') && !!defined?(::RedisClient)
+    HAS_REDIS_CLIENT = defined?(::Redis) &&
+      Gem::Version.new(::Redis::VERSION) >= Gem::Version.new('5.0.0') &&
+      !!defined?(::RedisClient)
```

Comment:
This constant is loaded by a simple `require` of the class performed by `test/new_relic/agent/instrumentation/instrumentation_test.rb`, which runs without the `redis` gem. We need the `defined?(::Redis)` guard to prevent errors.